drivers/sensor/si1153/si1153.c

```diff
@@ -0,0 +1,655 @@
+
+/*
+ * Copyright (c) 2017 IpTronix S.r.l.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* @file
+ *
+ * Driver for Silicon Labs Proximity/Ambient Light Sensor
+ *
+ */
+#include <kernel.h>
+#include <string.h>
+#include <sensor.h>
+#include <init.h>
+#include <gpio.h>
+#include <misc/byteorder.h>
+#include <misc/__assert.h>
+#include <i2c.h>
+
+#include "si1153.h"
+
+struct si115x_samples {
+	u32_t timestamp;         /* Timestamp to record */
+	u8_t irq_status;
+	s32_t ch0;
+	s32_t ch1;
+	s32_t ch2;
+	s32_t ch3;
+};
+
+static struct si1153_data si1153_data;
+
+struct k_thread si1153_sensor_thread;
+
+static int si1153_reg_read(struct si1153_data *data, u8_t reg)
+{
+	u8_t buf[1];
+	int ret;
+
+	ret = i2c_burst_read(data->i2c_master, data->i2c_slave_addr,
+			     reg, buf, 1);
+	if (ret) {
+		return ret;
+	}
+
+	return buf[0];
+}
+
+static int si1153_reg_write(struct si1153_data *data, u8_t reg, u8_t val)
+{
+	return i2c_reg_write_byte(data->i2c_master, data->i2c_slave_addr,
+				  reg, val);
+}
+
+static int si1153_block_read(struct si1153_data *data, u8_t start,
+			     int size, uint8_t *buf)
+{
+	return i2c_burst_read(data->i2c_master, data->i2c_slave_addr,
+			      start, buf, size);
+}
+
+static int si1153_block_write(struct si1153_data *data, u8_t start,
+			      int size, uint8_t *buf)
+{
+	return i2c_burst_write(data->i2c_master, data->i2c_slave_addr,
+			       start, buf, size);
+}
+
+static int _wait_until_sleep(struct si1153_data *data)
+{
+	int retval;
+	u8_t count = 0;
+
+	/* This loops until the Si115x is known to be in its sleep state
+	 * or if an i2c error occurs
+	 */
+	while (count < 5) {
```

Comment:
Maybe define a constant instead of using the number 5. This applies to multiple places in this file.